app/policies/policy_context.rb

```diff
@@ -0,0 +1,12 @@
+class PolicyContext
+  attr_reader :record
+
+  def initialize(record, user)
+    @record = record
+    @user = user
+  end
+
+  def policy_class
+    "#{@user.roles.uniq.first}DefenceRequestPolicy".classify
```

Comment:
I agree this needs looking at in more detail but probably not here. Ill stick a story in the icebox